lisa/features/gpu.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from enum import Enum
+from typing import Any, cast
+
+from lisa.feature import Feature
+from lisa.operating_system import Linux, Redhat, Ubuntu
+from lisa.tools import Uname
+from lisa.util import LisaException
+
+FEATURE_NAME_GPU = "Gpu"
+
+ComputeSDK = Enum(
+    "ComputeSDK",
+    [
+        # GRID Driver
+        "GRID",
+        # CUDA Driver
+        "CUDA",
+    ],
+)
+
+# Link to the latest GRID driver
+# The DIR link is
+# https://download.microsoft.com/download/9/5/c/95c667ff-ab95-4c56-89e0-e13e9a76782d/NVIDIA-Linux-x86_64-460.32.03-grid-azure.run
+GRID_DRIVER = "https://go.microsoft.com/fwlink/?linkid=874272"
+
+
+class Gpu(Feature):
+    def __init__(self, node: Any, platform: Any) -> None:
+        super().__init__(node, platform)
+        self._log = self._node.log
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_GPU
+
+    # download and install NVIDIA grid driver
+    def _install_grid_driver(self, version: str) -> None:
+        self._log.info("Starting GRID driver installation")
+        if not version.strip():
+            version = GRID_DRIVER
+
+        # grid_filename = "NVIDIA-Linux-x86_64-grid.run"
+
+        # download and install the NVIDIA GRID driver
+
+    # download and install CUDA Driver
+    def _install_cuda_driver(self, version: str) -> None:
+        self._log.info("Starting CUDA driver installation")
+        cuda_repo = ""
+        linux_os: Linux = cast(Linux, self._node.os)
+
+        if not version.strip():
+            version = "10.1.105-1"
+
+        # CUDA driver installation for redhat distros
+        if isinstance(self._node.os, Redhat):
+            cuda_repo_pkg = f"cuda-repo-rhel7-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel7/x86_64/{cuda_repo_pkg}"
+            )
+            linux_os = Redhat(self._node)
```

Comment:
I'm confused: isn't OS resolving something already taken care of if the node is initialized? Is not the case here?

Reply:
outdated iteration. I think you are not looking at latest code.

Reply:
I thought I was, but then.. surprise! A fixup commit in the chain :)

Reply:
oh well I don't see it in the latest commit. Yes let me squash it in my next push.